server/container/util-java8/src/main/java/org/apache/james/util/OptionalUtils.java

```diff
@@ -45,4 +45,17 @@
             .filter(Optional::isPresent)
             .orElse(optional2);
     }
+
+    public static <T> boolean containsDifferent(Optional<T> requestValue, T storeValue) {
+        return requestValue
+            .filter(value -> !value.equals(storeValue))
+            .isPresent();
+    }
+
+    public static <T> boolean contentChangedCompareToStoreValue(Optional<T> requestValue, Optional<T> storeValue) {
```

Comment:
It is better but I still think it is weird

Reply:
Actually why not removing this function and replace the call for it with requestValue.isPresent() && requestValue.equals(storeValue).

As a reader I will think that way more readable than this obscure ``contentChangedCompareToStoreValue``

Reply:
In fact I did it but it make the code on *Processor is weird. It's better when we move those function to separated helper class.